### PR TITLE
Dmyterko yuriy

### PR DESCRIPTION
**Pull request by ydmyterko**
Added following homework: HM1, HM2, HM3, HM4, HM5

### CODE REVIEW

File: ydmyterko/HM1/hm1.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name = input("WHAT IS YOUR NAME? ")
```
Comment (kolyasalubov) [inline ydmyterko/HM1/hm1.py:1]: good job

File: ydmyterko/HM1/hm2.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+A = int(input("Enter A value: "))
```
Comment (kolyasalubov) [inline ydmyterko/HM1/hm2.py:1]: good job

File: ydmyterko/HM3/HW1.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+file = open("./Zen.txt", "r")
```
Comment (kolyasalubov) [inline ydmyterko/HM3/HW1.py:1]: good job

File: ydmyterko/HM3/HW2.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+number = input("Enter 4 digits number:")
```
Comment (kolyasalubov) [inline ydmyterko/HM3/HW2.py:1]: good job

File: ydmyterko/HM3/HW3.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#solution 1
+x = 1
```
Comment (kolyasalubov) [inline ydmyterko/HM3/HW3.py:2]: good job

File: ydmyterko/HM4/HM1.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+number = int(input('Please enter a number: '))
```
Comment (kolyasalubov) [inline ydmyterko/HM4/HM1.py:1]: good job

File: ydmyterko/HM5/HW1.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+list_digits = [1, 2, 4, 9]
```
Comment (kolyasalubov) [inline ydmyterko/HM5/HW1.py:1]: good job

File: ydmyterko/HM5/HW2.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+number = int(input("Please enter a number bigger then 1: "))
```
Comment (kolyasalubov) [inline ydmyterko/HM5/HW2.py:1]: good job

File: ydmyterko/HM6/HW1.py [hunk 9]
```diff
@@ -0,0 +1,16 @@
+divisible_by_2 = []
+divisible_by_3 = []
+divisible_by_2_and_3 = []
+
+for i in range(1,10):
+    print(i)
```
Comment (kolyasalubov) [inline ydmyterko/HM6/HW1.py:6]: must be without this line

File: ydmyterko/HM6/HW2.py [hunk 10]
```diff
@@ -0,0 +1,9 @@
+attempt=0
```
Comment (kolyasalubov) [inline ydmyterko/HM6/HW2.py:1]: good job

File: ydmyterko/HM1/hm1.py (hunk 1 above)
Comment (ydmyterko) [reply ydmyterko/HM1/hm1.py:1]: thanks

File: ydmyterko/HM3/HW1.py (hunk 3 above)
Comment (ydmyterko) [reply ydmyterko/HM3/HW1.py:1]: thanks

File: ydmyterko/HM3/HW2.py (hunk 4 above)
Comment (ydmyterko) [reply ydmyterko/HM3/HW2.py:1]: thanks

File: ydmyterko/HM3/HW3.py (hunk 5 above)
Comment (ydmyterko) [reply ydmyterko/HM3/HW3.py:2]: thanks

File: ydmyterko/HM5/HW1.py (hunk 7 above)
Comment (ydmyterko) [reply ydmyterko/HM5/HW1.py:1]: thanks

File: ydmyterko/HM6/HW1.py (hunk 9 above)
Comment (ydmyterko) [reply ydmyterko/HM6/HW1.py:6]: thanks

File: ydmyterko/HM4/HM1.py (hunk 6 above)
Comment (ydmyterko) [reply ydmyterko/HM4/HM1.py:1]: thanks

Comment (ydmyterko) [reply ydmyterko/HM4/HM1.py:1]: thanks

File: ydmyterko/HM5/HW2.py (hunk 8 above)
Comment (ydmyterko) [reply ydmyterko/HM5/HW2.py:1]: thanks

File: ydmyterko/HM6/HW2.py (hunk 10 above)
Comment (ydmyterko) [reply ydmyterko/HM6/HW2.py:1]: thanks

File: ydmyterko/HM7/10_boolean_to_str.py [hunk 11]
```diff
@@ -0,0 +1,8 @@
+def bool_to_word(boolean):
```
Comment (kolyasalubov) [inline ydmyterko/HM7/10_boolean_to_str.py:1]: good job

File: ydmyterko/HM7/11_counting_sheeps.py [hunk 12]
```diff
@@ -0,0 +1,14 @@
+# array1 = [True,  True,  True,  False,
+#                   True,  True,  True,  True ,
+#                   True,  False, True,  False,
+#                   True,  False, False, True ,
+#                   True,  True,  True,  True ,
+#                   False, False, True,  True ];
+
+def count_sheeps(sheep):
```
Comment (kolyasalubov) [inline ydmyterko/HM7/11_counting_sheeps.py:8]: good job

File: ydmyterko/HM7/12_whos_tail.py [hunk 13]
```diff
@@ -0,0 +1,9 @@
+def correct_tail(body, tail):
```
Comment (kolyasalubov) [inline ydmyterko/HM7/12_whos_tail.py:1]: good job

File: ydmyterko/HM7/1_greet.py [hunk 14]
```diff
@@ -0,0 +1,15 @@
+# Solution #1
```
Comment (kolyasalubov) [inline ydmyterko/HM7/1_greet.py:1]: good job

File: ydmyterko/HM7/2_distance.py [hunk 15]
```diff
@@ -0,0 +1,7 @@
+import math
```
Comment (kolyasalubov) [inline ydmyterko/HM7/2_distance.py:1]: good job

File: ydmyterko/HM7/4_num_to_str.py [hunk 16]
```diff
@@ -0,0 +1,6 @@
+def number_to_string(num):
```
Comment (kolyasalubov) [inline ydmyterko/HM7/4_num_to_str.py:1]: good job

File: ydmyterko/HM7/5_reverse_words.py [hunk 17]
```diff
@@ -0,0 +1,6 @@
+def reverse(st):
```
Comment (kolyasalubov) [inline ydmyterko/HM7/5_reverse_words.py:1]: good job

File: ydmyterko/HM7/6_reverce_list.py [hunk 18]
```diff
@@ -0,0 +1,5 @@
+def reverce_list(l):
```
Comment (kolyasalubov) [inline ydmyterko/HM7/6_reverce_list.py:1]: good job

File: ydmyterko/HM7/7_sum_of_digits.py [hunk 19]
```diff
@@ -0,0 +1,11 @@
+def solution(number):
```
Comment (kolyasalubov) [inline ydmyterko/HM7/7_sum_of_digits.py:1]: good job

File: ydmyterko/HM7/8_gas_calculation.py [hunk 20]
```diff
@@ -0,0 +1,7 @@
+def zero_fuel(distance_to_pump, mpg, fuel_left):
```
Comment (kolyasalubov) [inline ydmyterko/HM7/8_gas_calculation.py:1]: good job

File: ydmyterko/HM7/9_are_you_playing_banjo.py [hunk 21]
```diff
@@ -0,0 +1,5 @@
+def are_you_playing_banjo(name):
```
Comment (kolyasalubov) [inline ydmyterko/HM7/9_are_you_playing_banjo.py:1]: good job

File: ydmyterko/HM7/HW1.py [hunk 22]
```diff
@@ -0,0 +1,11 @@
+def func_larger_number(num1,num2):
```
Comment (kolyasalubov) [inline ydmyterko/HM7/HW1.py:1]: good job

File: ydmyterko/HM7/HW2.py [hunk 23]
```diff
@@ -0,0 +1,38 @@
+def Rectangle_area():
```
Comment (kolyasalubov) [inline ydmyterko/HM7/HW2.py:1]: good job

File: ydmyterko/HM7/HW3.py [hunk 24]
```diff
@@ -0,0 +1,9 @@
+def Number_of_characters(str) -> None:
```
Comment (kolyasalubov) [inline ydmyterko/HM7/HW3.py:1]: look at this implementation
def count_symbols_in_row(word):
    result = {}
    for item in word:
        if item in result:
            continue
        else:
            result.update({str(item): word.count(item)})
    return result

print(count_symbols_in_row("abracadabra"))

File: ydmyterko/HM7/3_noYelling [hunk 25]
```diff
@@ -0,0 +1,8 @@
+def filter_words(st):
```
Comment (kolyasalubov) [inline ydmyterko/HM7/3_noYelling:1]: good job

File: ydmyterko/HM8/HW1/main.py [hunk 26]
```diff
@@ -0,0 +1,4 @@
+from models import *
```
Comment (kolyasalubov) [inline ydmyterko/HM8/HW1/main.py:1]: good job

File: ydmyterko/HM8/HW2/PSW_validation.py [hunk 27]
```diff
@@ -0,0 +1,17 @@
+def psw_validation(password):
```
Comment (kolyasalubov) [inline ydmyterko/HM8/HW2/PSW_validation.py:1]: good job
look at this implementation:

import re

def validate_password(input_password):
    
    flag = True
    
    while flag:  
        if len(input_password)<6 or len(input_password)>16:
            break
        elif not re.search("[a-z]",input_password):
            break
        elif not re.search("[0-9]",input_password):
            break
        elif not re.search("[A-Z]",input_password):
            break
        elif not re.search("[$#@]",input_password):
            break
        # Returns a match where the string contains a white space character
        elif re.search("\s",input_password):  
            break
        else:
            return f"{input_password} is valid password"
            flag = False
            break

    if flag:
        return f"{input_password} is invalid password"

File: ydmyterko/HM8/HW3/area.py [hunk 28]
```diff
@@ -0,0 +1,19 @@
+import math
```
Comment (kolyasalubov) [inline ydmyterko/HM8/HW3/area.py:1]: good job

File: ydmyterko/HM8/HW3/main.py [hunk 29]
```diff
@@ -0,0 +1,20 @@
+import area
```
Comment (kolyasalubov) [inline ydmyterko/HM8/HW3/main.py:1]: good job

**Pull request merged**